Source/NSOperation.m

```diff
@@ -1050,7 +1093,24 @@ - (void) _thread
 	    {
 	      ENTER_POOL
               [NSThread setThreadPriority: [op threadPriority]];
+
+#if HAVE_DISPATCH_H || HAVE_DISPATCH_DISPATCH_H
+              dispatch_queue_t underlyingQueue = [self underlyingQueue];
+
+              if (nil != underlyingQueue)
+                {
+                  dispatch_sync(underlyingQueue, ^(void)
```

Comment:
in this case, underlyingQueue is the main queue right? Is there any chance this function gets called from the main thread? Because if so, this will deadlock.

Reply:
This method is always run in a background worker thread, so it won't ever be on any thread dispatch is using